### ISSUE TITLE
Question about perfomance and hyperparameters

### ISSUE THREAD

**Issue by Rodyarad** [opendilab/LightZero#450]
Hello, thanks for your work!

We have implemented a custom environment with discrete action spaces. We’ve observed that after reaching a certain level of performance (in terms of reward or success rate), the results begin to degrade during further training (we’re training with UniZero).

We have also encountered similar behavior with the original EfficientZero and EfficientZeroV2 repositories when running other custom environments.

Have you encountered performance degradation after a plateau? Are there any specific hyperparameters or strategies for solving it?

Thank you in advance for your response.

**Comment by puyuan1996** [comment]
Hello, thanks for your interest in LightZero and for your feedback!

We have indeed observed similar performance degradation issues with UniZero in the later stages of training (around 1M env steps). To address this, we have recently implemented several bug fixes and stability enhancements, which are detailed in https://github.com/opendilab/LightZero/pull/451.

Key improvements include:
1.  **KV Cache Fix**: We fixed a critical bug that caused the `kv_cache` to be improperly overwritten.
2.  **Policy Stability**: We introduced target-entropy control mechanisms and polished the stability of policy logits.

Currently, UniZero's performance on Qbert is much more stable than before (as shown in the attached figure). However, we still observe a drop around 2.6M steps, and we are actively working on further optimizations.

<img width="1280" height="766" alt="Image" src="https://github.com/user-attachments/assets/394679af-e8df-4cef-8b04-81c9f52bf227" />

<img width="1280" height="388" alt="Image" src="https://github.com/user-attachments/assets/9840b81c-7563-407e-8350-6c070996f3e9" />

If you would like to test these latest fixes, you can run the configuration file located at `zoo/atari/config/atari_unizero_segment_config.py` on the `dev-multitask-balance-clean-kvcachemanager` branch associated with the PR.

**Please Note**: The code in this branch is primarily for developing and verifying features for the ScaleZero paper. It is currently somewhat messy and unoptimized, but we will clean up and standardize it as soon as possible.

Regarding the other algorithms you mentioned: The MuZero version in the LightZero repository currently shows stable improvement up to 8M steps. 

<img width="1280" height="370" alt="Image" src="https://github.com/user-attachments/assets/8eaa3bd1-2920-4dae-93e2-8cc897472228" />

Since our EfficientZero (EZ) implementation is based on MuZero, we expect its performance to be similarly stable. We have not tested the long-run performance of the original EfficientZero and EfficientZeroV2 repositories.

We hope this helps.
